TKeazirian.HTTPServer.Tests/Router/RouterTests.cs

```diff
@@ -160,17 +160,35 @@ public void AddAllowedMethodsToListAddsMethods()
         Routes routes = new Routes();
         routes.AddRoute("/test_path", testRoute);
 
-        var allowedMethods = OptionsResponse.AddToAllowedMethodsForOptions(testRoute);
+        var allowedMethods = Router.AllowedMethodsWithHeadAndOptions(testRoute);
 
         Assert.Equal("GET, POST, HEAD, OPTIONS", allowedMethods);
     }
 
+    [Fact]
+    public void Returns501NotImplementedWhenMethodIsUnknownAkaNotInMethods()
+    {
+        Route testRoute = new Route(
+            new List<HttpMethod>() { HttpMethod.GET },
+            new MockHandler()
+        );
+        Routes routes = new Routes();
+        routes.AddRoute("/test_path", testRoute);
+
+        Request testRequest =
+            new Request(HttpMethod.UNKNOWN, "/test_path", "", "");
+        Router router = new Router(routes);
+
+        Response response = router.Route(testRequest);
+
+        Assert.Equal("HTTP/1.1 501 Not Implemented\r\n", response.ResponseStatusLine);
+    }
+
     [Theory]
-    [InlineData(HttpMethod.POST)]
-    [InlineData(HttpMethod.PUT)]
     [InlineData(HttpMethod.DELETE)]
+    [InlineData(HttpMethod.POST)]
     [InlineData(HttpMethod.PATCH)]
-    public void Returns501NotImplementedWhenMethodIsNotImplemented(HttpMethod method)
+    public void Returns405WhenMethodIsInEnumButNotSupportedForRoute(HttpMethod method)
```

Comment:
This name has the work `ENUM` in it; That's an implementation detail. What does `IsInEnum` mean in this context? Maybe rename the test to match that.

Reply:
I think this is a part of the [feedback from above](https://github.com/t-keazirian/HTTPServer/pull/30#discussion_r920324098) that was not fully resolved.

Reply:
Ah yes, I forgot I had mentioned that in the test name.  Thanks!

Reply:
The same goes for any supporting methods.  Who matters if it is an enum or not an enum? that should be abstracted away.